components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -1660,6 +1660,8 @@ public void invalidateAndCreateNewAccessToken(String oldAccessTokenId, String to
             // update new access token against authorization code if token obtained via authorization code grant type
             updateTokenIdIfAutzCodeGrantType(oldAccessTokenId, accessTokenDO.getTokenId(), connection);
 
+            IdentityDatabaseUtil.commitTransaction(connection);
```

Comment:
Seems like IdentityDatabaseUtil.commitTransaction(connection); already present at line 1671. This needs to be removed, otherwise there will be exceptions saying about transaction already commited.

Reply:
We need to commit the transaction here since postRefreshAccessToken() method needs the committed data from database. This commit was there previously before the code refactoring took place. Once we remove this commit operation here, refresh token process doesn't work for auth code grant type access token using the request object as we tested.

Reply:
There can be only one commit transaction for one begin transaction.
Also there can not be multiple transactions in the single flow.

Above comment means there are some more improvement to be done.

Reply:
Thank you very much for the insights, will do the needful ASAP.